crates/net/downloaders/src/metrics.rs

```diff
@@ -22,6 +22,15 @@ pub struct DownloaderMetrics {
     pub total_flushed: Counter,
     /// Number of items that were successfully downloaded
     pub total_downloaded: Counter,
+    /// The number of out-of-order requests sent by the downloader.
+    /// The consumer of the download stream is able to re-request data (headers or bodies) in case
+    /// it encountered a recoverable error (e.g. during insertion).
+    /// Out-of-order request happen when:
+    ///     - the headers downloader `SyncTarget::Tip` hash is different from the previous sync
+    ///       target hash.
+    ///     - the new download range start for bodies donwloader is less than the last block number
```

Comment:
I'm still confused by the naming, sorry for being pedantic here, but "out-of-order" is really hard for me to understand and doesn't really fit here imo.
But I don't really have any suggestions besides making it clear that these are requests sent because the target(range) changed.

Reply:
the reasoning behind naming them out-of-order was that the downloaders start at a specific point and request data in a certain direction moving forward (asc or desc). any request for data before the point that was already passed i named `out-of-order`. anyway, how about `duplicate_data_requests`, `range_recovery_requests` or `repeated_requests`?

Reply:
ah okay, right, yeah this makes sense then.
because `order` would be falling for headers and out-of-order would be a higher block, and for bodies vice versa